packages/extension-ui/src/Popup/index.tsx

```diff
@@ -114,39 +116,41 @@ export default function Popup (): React.ReactElement {
     : Welcome;
 
   return (
-    <Loading>{accounts && authRequests && metaRequests && signRequests && (
-      <ActionContext.Provider value={_onAction}>
-        <SettingsContext.Provider value={settingsCtx}>
-          <AccountContext.Provider value={accountCtx}>
-            <AuthorizeReqContext.Provider value={authRequests}>
-              <MediaContext.Provider value={cameraOn && mediaAllowed}>
-                <MetadataReqContext.Provider value={metaRequests}>
-                  <SigningReqContext.Provider value={signRequests}>
-                    <ToastProvider>
-                      <Switch>
-                        <Route path='/account/create'><CreateAccount /></Route>
-                        <Route path='/account/forget/:address'><Forget /></Route>
-                        <Route path='/account/export/:address'><Export /></Route>
-                        <Route path='/account/import-qr'><ImportQr /></Route>
-                        <Route path='/account/import-seed'><ImportSeed /></Route>
-                        <Route path='/account/restore-json'><RestoreJson /></Route>
-                        <Route path='/account/derive/:address/locked'><Derive isLocked /></Route>
-                        <Route path='/account/derive/:address'><Derive /></Route>
-                        <Route
-                          exact
-                          path='/'
-                        >
-                          <Root />
-                        </Route>
-                      </Switch>
-                    </ToastProvider>
-                  </SigningReqContext.Provider>
-                </MetadataReqContext.Provider>
-              </MediaContext.Provider>
-            </AuthorizeReqContext.Provider>
-          </AccountContext.Provider>
-        </SettingsContext.Provider>
-      </ActionContext.Provider>
-    )}</Loading>
+    <ErrorBoundary FallbackComponent={ErrorFallback}>
```

Comment:
The issue with only having it this far up the tree is that you can never recover. So it would need to sit inside the signing component and still have a "cancel" when it gets into this state.

However... probably need a reproduction to see it working. (Well, I guess we can make it crash on purpose in the affected parts)

Reply:
Right, having it this high in the tree was a first attempt so try to see more helpful errors.
I'll play with it lower.

Reply:
100%, the first round is just catching... hopefully :)